doc/user/content/manage/troubleshooting.md

```diff
@@ -10,77 +10,309 @@ aliases:
+Others are specific to differential dataflow (`Get`, `ArrangeBy`).
 
+In general, a high level understanding of what these operators do is sufficient for effective debugging:
+`Filter` filters records, `Join on` joins records from two or more inputs, `Map` applies a function to transform records, etc.
+The `ArrangeBy` operator indexes data and stores it in memory for fast point lookups.
+It's not important to have a deep understanding of all these operators for effective debugging, though.
+
+Behind the scenes, the operator graph is turned into a dataflow that can be evaluated by the Differential Dataflow runtime.
+The dataflow is organized in a hierarchical structure. 
+It contains operators and so-called regions, which in turn contain operators and (sub)regions.
+
+In our example, the dataflow contains a InputRegion and a BuildRegion.
+And the BuildRegion contains the operators from the above plan.
+
+![Regions and operator visualization](/images/regions-and-operators.png)
+
+Again, it's not too important for our purposes to understand what these regions do and how they are used to structure the operator graph.
+For our purposes it's merely important to know than that they define a hierarchy on the operators.
+
+
+
+## The system catalog and introspection relations
+
+Materialize collects a lot of useful information about the dataflows and operators in the System Catalog in [introspection relations](/sql/system-catalog/mz_internal/#replica-introspection-relations).
+The introspection relations are extremely valuable to troubleshoot and to understand what is happening under the hood,
+in particular if Materialize is not behaving in the expected way.
+However, it is important to understand that most of the statistics we need for troubleshooting purposes are specific to the cluster that is running the queries we want to debug.
+
+{{< warning >}}
 The contents of introspection relations differ depending on the selected cluster and replica.
 As a consequence, you should expect the answers to the queries below to vary depending on which cluster you are working in.
 In particular, indexes and dataflows are local to a cluster, so their introspection information will vary across clusters.
+{{< /warning >}}
 
 <!--
 [//]: # "TODO(joacoc) We should share ways for the user to diagnose and troubleshoot if and how fast a source is consuming."
 ``` -->
 
-## Why is Materialize running so slowly?
+## Where is Materialize spending time on computations?
 
 Materialize spends time in various dataflow operators maintaining
 materialized views or indexes. If Materialize is taking more time to update
 results than you expect, you can identify which operators
 take the largest total amount of time.
 
+### Finding expensive queries/dataflows
+
+To understand which query or dataflow, respectively, is taking the most time we can query the `mz_scheduling_elapsed` relation.
```

Comment:
"Which query or dataflow, respectively". What is this making reference to? `mz_scheduling_elapsed` is bound to a relation, and in the following section you dig into operators per dataflow, so I'm missing the bit where we make references to queries.

Reply:
Maybe the confusion here comes from unclarity about how SQL constructs map to compute constructs?

* index -> dataflow (with an arrangement at the end)
* materialized view -> dataflow (with a storage sink at the end)
* view -> doesn't touch compute directly
* SELECT query -> either:
  * peek against an arrangement ("fast path") _or_
  * temporary dataflow (with an arrangement at the end) + peek against that arrangement